spec/controllers/labels_controller_spec.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe LabelsController do
+  render_views
+
+  let(:user) { create(:user) }
+  let(:label) { create(:label) }
+
+  describe 'GET #search' do
+    context 'without being signed in' do
+      it 'redirects to other page' do
+        get :search, params: {search: label.name, page: 1}
+        expect(response.body).to match(%r{^<html><body>You are being.*redirected.*</body></html>$})
+      end
+    end
+
+    context 'with valid params' do
+      subject(:request) { get :search, params: {search: label.name, page: 1} }
+
+      before { sign_in user }
+
+      it 'returns valid json' do
+        request
+        expect(response.body).to match(/^{"results":\[.*\],"pagination":{"more":(true|false)}}$/)
```

Comment:
While the test suggests that you are checking for a valid JSON, this is not exactly true. You are checking that the response matches a given pattern, which will match a valid JSON, but there are also constellations, where an invalid JSON would still match. "Just" matching for a valid JSON would be performed by trying to parse the JSON and checking for any parse errors. However, I doubt this test would be useful, since generating and parsing the JSON is mostly performed by libraries we haven't written ourselves (and thus the test would cover the library functionality...). Therefore, ask yourself: What do you _actually_ want to test here? I think you had a good intention, but just need to rework the test description `returns valid json` and the expectations.

For the expectations, the following might come handy: `response.parsed_body` will give you a hash representation of the JSON, allowing you to specify more fine granular tests. For example, you could check for some predefined keys to be present in the response or some data types (`results` should be an array, ...)

Reply:
I changed the expectations to check for keys and variable types.